### DISCUSSION TITLE
CreateModel in Parallel.For  need more time in 6.18.1 version,but ok in 6.0.0

### DISCUSSION THREAD

**Discussion by mypursuit** [rabbitmq/rabbitmq-dotnet-client#1544] [General]
### Describe the bug

server：RabbitMQ 3.5.6, Erlang 18.1
I wanto use `ConsumerDispatchConcurrency` property，but not support in 6.0.0 version.
when I upgrade to 6.8.1 version to test my code,I found newer version client need more time in Parallel.For.
Test result like this:
- 6.0.0 version :114ms
- 6.8.1 version :13785ms

### Reproduction steps

```
ConnectionFactory factory = new()
{
    //  6.0.0
    Uri = new Uri("amqp://rabbit:1234@192.168.11.51:5672/"),
    RequestedConnectionTimeout = TimeSpan.FromSeconds(60),
    RequestedHeartbeat = TimeSpan.FromSeconds(60),//心跳时间，秒

    // 5.1.2、5.2.0 版本个客户端初始化属性
    //Uri = new Uri("amqp://rabbit:1234@192.168.11.51:5672/"),
    //RequestedConnectionTimeout = 60 * 1000,
    //RequestedHeartbeat = 60,

    // 3.6.9、4.1.3 版本客户端初始化属性
    //Uri = "amqp://rabbit:1234@192.168.11.51:5672/",
    //RequestedConnectionTimeout = 60 * 1000,
    //RequestedHeartbeat = 60,

    //  公有
    // RequestedChannelMax = _MaxChannel,// 每个连接的最大信道连接数
    AutomaticRecoveryEnabled = true,//自动重连
    NetworkRecoveryInterval = TimeSpan.FromSeconds(10),//网络故障恢复间隔时间
    
};
IConnection conn = factory.CreateConnection();
var x = conn.CreateModel();
String str;
Stopwatch sw = Stopwatch.StartNew();
/**
    * 测试并发任务逻辑：一次启1000个task通知运行，在6.0+版本会存在operate timeout异常，分析是由于task同时起了，但相互等待出问题
        // 每次构建一个channel；超过100，巨慢
        List<Task> tasks = MultiTask.Build(1000, index =>
        {
            // var x = factory.CreateConnection().CreateModel();
            conn.CreateModel();
        });
        Task.WaitAll(tasks.ToArray());
    换成 Parallel.For；和MultiTask区别在于，不是一次把所有task先构建出来
    */
Parallel.For(0, 200, (index) =>
{
    var x = conn.CreateModel();
});

sw.Stop();
str = sw.ElapsedMilliseconds.ToString();
```
### Expected behavior

why has the different,How to solve it ,thank you

**Comment by michaelklishin** [comment]
@mypursuit you are welcome to investigate what's taking the time with a profiler and post your findings.

I'm all for making the client more efficient but this is not a typical or reasonable workload. It is rare to see channels opened in parallel like that because if optimizing this part matters, it means that [short-lived channels are used](https://www.rabbitmq.com/docs/channels#lifecycle), which is an explicitly recommended against practice.

**Reply by mypursuit** [reply]
I understandl ; but in a concurrent environment, it is not possible to share a model.
I do not understand in 6.0.0 version,it is work well

**Comment by lukebakken** [comment]
> ,I found newer version client need more time in Parallel.For

Increase your ThreadPool size.

https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/860#issuecomment-643262741

https://github.com/lukebakken/rabbitmq-dotnet-client-860/blob/main/Program.cs

@mypursuit - please note that version 7 fixes this issue. If you could test the latest alpha release I would **GREATLY** appreciate it:

https://www.nuget.org/packages/RabbitMQ.Client/7.0.0-alpha.5

**Reply by mypursuit** [reply]
@lukebakken I am glad to test this in 7.0.0-alpha.5 version 
---
The test result is `152ms`; even i send message in the action，just only spend  `211ms` .
This excites me because the new versions are all asynchronous, but I cannot use them in my production environment now; Looking forward to the official release!
---
```
String queueName = "FZJTest.rabbitmq2";
await (await conn.CreateChannelAsync()).QueueDeclareAsync(queueName, true, false, false, null);
await Parallel.ForAsync(0, 200, async (index, _) =>
 {
     var channel = await conn.CreateChannelAsync();
      var pros = new BasicProperties
      {
          ContentType = "text/plain",
          ContentEncoding = "utf-8",
          DeliveryMode = DeliveryModes.Persistent,
      };
      byte[] body = Encoding.UTF8.GetBytes("11111111111111111111111111111111111");
      await channel.BasicPublishAsync<BasicProperties>(String.Empty, queueName, pros, body, false);
 });

sw.Stop();
str = sw.ElapsedMilliseconds.ToString();
```

**Reply by lukebakken** [reply]
Please keep testing alpha, beta, and RC releases as they are available. It helps a LOT.

**Comment by mypursuit** [comment]
Thank you every much,it is useful to me❤️



---- Replied Message ----
| From | Luke ***@***.***> |
| Date | 04/25/2024 23:52 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [rabbitmq/rabbitmq-dotnet-client] CreateModel in Parallel.For need more time in 6.18.1 version,but ok in 6.0.0 (Discussion #1544) |

Closed #1544 as resolved.
